src/main/java/at/hannibal2/skyhanni/features/mining/crystalhollows/CrystalNucleusTracker.kt

```diff
@@ -0,0 +1,192 @@
+            description = "Resets the Crystal Nucleus Tracker"
+            category = CommandCategory.USERS_RESET
+            callback { resetCommand() }
+        }
+    }
+
+    @HandleEvent(priority = HIGH)
+    fun onCrystalNucleusLoot(event: CrystalNucleusLootEvent) {
+        addCompletedRun()
+        for ((itemName, amount) in event.loot) {
+            // Gemstone and Mithril Powder
+            if (itemName.contains(" Powder")) continue
+            NEUInternalName.fromItemNameOrNull(itemName)?.let {
+                tracker.addItem(it, amount, false)
+            }
+        }
+    }
+
+    private fun addCompletedRun() {
+        tracker.modify {
+            it.runsCompleted++
+        }
+    }
+
+    private fun drawDisplay(data: Data): List<Searchable> = buildList {
+        addSearchString("§e§lCrystal Nucleus Profit Tracker")
+        var profit = tracker.drawItems(data, { true }, this)
+
+        val runsCompleted = data.runsCompleted
+
+        if (runsCompleted > 0) {
+            val jungleKeyCost = jungleKeyItem.getPrice() * runsCompleted
+            profit -= jungleKeyCost
+            val jungleKeyCostFormat = jungleKeyCost.shortFormat()
+            add(
+                Renderable.hoverTips(
+                    " §7${runsCompleted}x §5Jungle Key§7: §c-$jungleKeyCostFormat",
+                    listOf(
+                        "§7You lost §c$jungleKeyCostFormat §7of total profit",
+                        "§7due to §5Jungle Keys§7."
+                    ),
+                ).toSearchable(),
+            )
+
+            val robotPartsCost = robotPartItems.sumOf { it.getPrice() } * runsCompleted
+            profit -= robotPartsCost
+            val robotPartsCostFormat = robotPartsCost.shortFormat()
+            add(
+                Renderable.hoverTips(
+                    " §7${runsCompleted * 6}x §9Robot Parts§7: §c-$robotPartsCostFormat",
+                    listOf(
+                        "§7You lost §c$robotPartsCostFormat §7of total profit",
+                        "§7due to §9Robot Parts§7."
+                    ),
+                ).toSearchable(),
+            )
+
+            add(
+                Renderable.hoverTips(
+                    "§7Runs completed: §e${runsCompleted.addSeparators()}",
+                    listOf("§7You completed §e${runsCompleted.addSeparators()} §7Crystal Nucleus Runs."),
+                ).toSearchable(),
+            )
+
+            add(tracker.addTotalProfit(profit, data.runsCompleted, "run"))
+        }
+
+        tracker.addPriceFromButton(this)
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent) {
+        if (!isEnabled()) return
+        tracker.renderDisplay(config.position)
+    }
+
+    @SubscribeEvent
+    fun onIslandChange(event: IslandChangeEvent) {
+        if (event.newIsland == IslandType.CRYSTAL_HOLLOWS) {
+            tracker.firstUpdate()
+        }
+    }
+
+    fun resetCommand() {
```

Comment:
no need to add this function, just call `tracker.resetCommand()` inside the command callback